### ISSUE TITLE
Details about snpe config json

### ISSUE THREAD

**Issue by MadanKrishnan97** [microsoft/vision-ai-developer-kit#326]
Hi, I need more information about the config json for snpe converter. I want to convert a Tensorflow model to the dlc using convert_tf_model. I have referred to https://azure.github.io/Vision-AI-DevKit-Pages/docs/model/.

For easier comparison and understanding, let us consider the following https://github.com/microsoft/vision-ai-developer-kit/blob/master/machine-learning-notebooks/models/mobilenet-soda-cans/orig/va-snpe-engine-library_config.json

- Engine
I understand that 0 and 1 for single and multiple output respectively. But when is the output considered multiple? If I am detecting 2 or more objects is it multiple? Is 'EngineOutput' the same as 'Engine'?

- NetworkIO
I understand that 0 and 1 are for UserBuffer and ITensor. When do I use what?

- InputLayers
Is it just the name of the input layer?

- OutputLayers
Is it again the name of the output layer?

- ResultLayers
I am unable to understand the difference between OutputLayers and ResultLayers

To give more of a context, I have a custom tensorflow model in the form of a .pb file. There is one input (inp) and 2 outputs (out1, out2). My config looks as shown.

```json
{
"Engine":1,
"NetworkIO":1,
"ScaleWidth":260,
"ScaleHeight":260,
"PixelNorm":127.5,
"BlueMean":104,
"GreenMean":117,
"RedMean":123,
"TargetFPS":10,
"ConfThreshold":0.3,
"DLC_NAME":"model.dlc",
"LABELS_NAME":"output_labels.txt",
"InputLayers":"inp:0",
"OutputLayers":["out1", "out2"],
"ResultLayers":["out1:0", "out2:0"],
"Runtime":1
}
```

Is there a more detailed explanation of the parameters in this json? The json works for the models available in the git repo, but for custom models it is hard to figure out what exactly is expected. Thanks!

**Comment by MS-Nikunja** [comment]
@MadanKrishnan97, allow us some time to figure out and we will get back to you.

**Comment by MadanKrishnan97** [comment]
@MS-Nikunja Sure thing.

**Comment by MS-Nikunja** [comment]
Hi @MadanKrishnan97, Refer this link to convert Tensorflow model to DLC: https://developer.qualcomm.com/docs/snpe/model_conv_tensorflow.html

**Comment by MadanKrishnan97** [comment]
Hi @MS-Nikunja. I have referred to the Qualcomm documentation before opening this issue. I was unable to find details about ResultLayers and OutputLayers among others.

**Comment by MS-Nikunja** [comment]
Hi @MadanKrishnan97, I am afraid there is no detailed explanation of the config.json other than https://azure.github.io/Vision-AI-DevKit-Pages/docs/model/.

To answer your questions related to the parameters:
 
1. Engine: Yes, if your are detecting 2 or more objects, then the output is considered multiple. Also, 'EngineOutput' and 'Engine' are the same things.
 
2. Network inputs and outputs can be either user-backed buffers or ITensors (built-in SNPE buffers), but not both. The advantage of using user-backed buffers is that it eliminates an extra copy from user buffers to create ITensors. You can use any one of them. Both methods of loading network inputs are shown in detail in this link: https://developer.qualcomm.com/docs/snpe/cplus_plus_tutorial.html

3. InputLayers and OutputLayers are names of input layer and output layer respectively.

4. The difference in OutputLayer and ResultLayer (used for comapring the results)  is mentioned in the parameter description link mentioned by you. For more details on the Layers, you will need to dive deeper into Neural Networking.

**Comment by MadanKrishnan97** [comment]
@MS-Nikunja Thank you. I will close the issue
